engine/src/main/java/org/hibernate/validator/cfg/ConstraintMapping.java

```diff
@@ -22,9 +26,24 @@
 	 *
 	 * @param <C> The type to be configured.
 	 * @param beanClass The bean class on which to define constraints. All constraints defined after calling this method
-	 * are added to the bean of the type {@code beanClass} until the next call of {@code type}.
+	 * are added to the bean of the type {@code beanClass} until the next call of {@code type} or {@code annotation}.
 	 *
 	 * @return Instance allowing for defining constraints on the specified class.
 	 */
 	<C> TypeConstraintMappingContext<C> type(Class<C> beanClass);
+	
+	/**
+	 * Starts defining {@link javax.validation.ConstraintValidator}s to be executed for the specified constraint (i.e. annotation class).
+	 * Each constraint may only be configured once within all constraint mappings used for configuring one validator
+	 * factory.
+	 *
+	 * @param <A> The annotation type to be configured.
+	 * @param annotationClass The annotation class on which to define the validators. This type must be an
+	 * {@code @interface} annotated with {@code javax.validation.Constraint}. All validators defined after calling
+	 * this method are added to the annotation of the type {@code annotationClass} until the next call
+	 * of {@code type} or {@code annotation}.
+	 *
+	 * @return Instance allowing for defining validators to be executed for the specified constraint.
+	 */
+	<A extends Annotation> ConstraintDefinitionContext<A> constraint(Class<A> annotationClass);
```

Comment:
I'd name it `constraintDefinition()` to differentiate from actually applying constraints via the existing `constraint()` method and be more inline with the XML names.